### ISSUE TITLE
TF/Keras RNN initial_state with stateful = false

### ISSUE THREAD

**Issue by junja94** [tensorflow/tensorflow#11557]
I'm trying to put different initial hidden state into recurrent layers. I saw many issues about initializing states using keras and most of the solutions were using layer.reset_state(state). But in my application, I have to set stateful = False and I cannot use reset_state. As it is written in the document, I tried initial_state argument in calling layers. 

Using the code below, I compared the outputs and the resulting outputs were always the same. I tried different initializers and different initial states but the both outputs were the same.

**When the stateful is False, is there other possible way to set initial hidden state???**

for size in dimension[1:]:
    initial_ = tf.ones(shape=[batchnum, size])
    recurrent_layer = GRU(units=size, return_sequences=True, kernel_initializer=one,
                                 recurrent_initializer=one)
    recurrent_layertest = GRU(units=size, return_sequences=True, kernel_initializer=one,
                                 recurrent_initializer=one)

    x = recurrent_layer(inputs=x)
    xtest = recurrent_layertest(inputs=xtest, initial_state = initial_)

In my last issue https://github.com/tensorflow/tensorflow/issues/11553, there was a suggestion to put inputs as [state, initial state] and It didn't work saying GRU has no attribute 'states'. I think this happens because the stateful is False.

**Comment by drpngx** [comment]
@fchollet any idea? Is that a documentation issue?

**Comment by fchollet** [comment]
> When the stateful is False, is there other possible way to set initial hidden state???

You can use `reset_states([state1, state2, ...])` to set states *by value* (e.g. by specifying numpy arrays). To specify *symbolic* states, use the `initial_state` argument when calling the layer instance on new inputs.

**Comment by junja94** [comment]
@fchollet   I tried as you suggested, and it didn't work.
The code I used is as below. I tested several times with different inputs.
I think the problem is that the function ignores initial_state argument in   **xtest = net.layers[idx+1](inputs = xtest, initial_state = initial_)**  I tried putting weird shaped initial states (like [1, 1000000]) and there was no error.
Could you check this code please?? Please let me know if I'm using it wrong.

Thanks


      self.input = tf.placeholder(dtype, shape=[None, None, dimension[0]],
                                    name=fn.input_names[0])  # Batch, timelen, dim
        testinput = tf.placeholder(dtype, shape=[None, None, dimension[0]],
                                   name='testin')
        # Network
        input_layer = layers.InputLayer(input_tensor=self.input, dtype=dtype)
        input_layertest = layers.InputLayer(input_tensor=testinput, dtype=dtype)

        x = input_layer.output
        xtest = input_layertest.output

        num_rnn = 0

        batchnum = tf.shape(self.input)[0]
        for size in dimension[1:]:
            recurrent_layer = layers.GRU(units=size, return_sequences=True, trainable = True)
            x = recurrent_layer(inputs=x)
            num_rnn += 1

        output = x
        net = models.Model(inputs=self.input, outputs=output)

# Initial_state 

        for idx,size in enumerate(dimension[1:]):
            initial_ = tf.random_uniform(shape=[batchnum, size*100000])
            xtest = net.layers[idx+1](inputs = xtest, initial_state = initial_)

        outputtest = xtest
        testnet = models.Model(inputs=testinput, outputs=outputtest)
